File: shortest_common_supersequence.cpp
//{ Driver Code Starts
//Initial template for C++

#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++


class Solution
{
    int dp[101][101];
    int M, N;
    string x, y;
    
    int solve(int i, int j) {
        if (i == M && j == N) return 0;
        if (i == M) return N - j;
        if (j == N) return M - i;
        if (dp[i][j] != -1) return dp[i][j];
        if (x[i] != y[j]) {
            dp[i][j] = min(solve(i+1, j), solve(i, j+1)) + 1;
        } else {
            dp[i][j] = solve(i+1, j+1) + 1;
        }
        //cout << i << ' '<< j << ' ' << dp[i][j] << endl;
        return dp[i][j];
    }
    public:
    //Function to find length of shortest common supersequence of two strings.
    int shortestCommonSupersequence(string X, string Y, int m, int n)
    {
        //code here
        M = m;
        N = n;
        x = X;
        y = Y;
        memset(dp, -1, sizeof(dp));
        return solve(0, 0);
    }
};

//{ Driver Code Starts.

int main()
{   
    
    int t;
    
    //taking total testcases
    cin >> t;
    while(t--){
    string X, Y;
    //taking String X and Y
	cin >> X >> Y;
	
	//calling function shortestCommonSupersequence()
	Solution obj;
	cout << obj.shortestCommonSupersequence(X, Y, X.size(), Y.size())<< endl;
    }
	return 0;
}


// } Driver Code Ends

File: dp/shortest_common_supersequence.py
# User function Template for python3


class Solution:
    # Function to find length of shortest common supersequence of two strings.
    def shortestCommonSupersequence(self, X, Y, m, n):
        # code here
        dp = [[0 for _ in range(n + 1)] for _ in range(m + 1)]
        for i in range(m + 1):
            dp[i][0] = i
        for i in range(n + 1):
            dp[0][i] = i
        for i in range(m):
            for j in range(n):
                if X[i] == Y[j]:
                    dp[i + 1][j + 1] = dp[i][j] + 1
                else:
                    dp[i + 1][j + 1] = min(dp[i][j + 1], dp[i + 1][j]) + 1

        return dp[m][n]


# {
# Driver Code Starts
# Initial Template for Python 3

# contributed by RavinderSinghPB
if __name__ == "__main__":
    t = int(input())
    for tcs in range(t):
        X, Y = input().split()

        print(Solution().shortestCommonSupersequence(X, Y, len(X), len(Y)))

# } Driver Code Ends
